### DISCUSSION TITLE
Octernship Assignment Submission Mistake

### DISCUSSION THREAD

**Discussion by Arindam-7** [community/community#58465] [GitHub Education]
### Hi everyone,

I had worked on the `feature` branch throughout the project, and as said in the instructions, I created a pull request to `main` branch. 
Everything was okay till that time, unless I overlooked the instructions and merged the PR myself.

So both of `main` and `feature` branches have final submission.

Idk I got confused, and created a few minor changes on the `feature` branch, and created a PR like the screenshot given below...

Have I committed huge mistake? 

<img width="969" alt="Screenshot 2023-06-19 at 6 41 42 PM" src="https://github.com/community/community/assets/63112908/c2804293-f428-426d-8806-13ad550514bd">

**Comment by abinashbordoloi** [comment]
"I thought the `main` branch was protected lol :'), 
You could have generated a pull request from a specific commit or the final commit in the `feature` branch and then generate a pull request for it to the `main` branch again, this could have provided a more detailed overview of the project. This would allow the reviewers to examine past commits and merges, enabling them to gain a deeper understanding of the work completed. 
Considering the commit history and incremental changes made during the development process should not make any problem :,).

**Reply by Arindam-7** [reply]
All the commits are visible on `main` branch 
<img width="303" alt="Screenshot 2023-06-20 at 10 44 02 AM" src="https://github.com/community/community/assets/63112908/aea686ba-af87-481c-b0fb-e804b454dcfb">


Also the `feedback` branch has all of the commit histories, and files changed visible
<img width="926" alt="Screenshot 2023-06-20 at 10 43 15 AM" src="https://github.com/community/community/assets/63112908/baea7bd4-c699-43e4-9c97-6e4046e5cddb">

And I've not deleted the `feature` branch as well. :)

I'm wondering if I should, as the other comment conveyed, reset the commits..cause deadline's already over, and it's risky as well!!


> You could have generated a pull request from a specific commit or the final commit in the feature branch and then generate a pull request for it to the main branch again
- I've actually done this, made minor changes in the feature branch, and left the PR open

**Comment by Neilblaze** [comment]
@Arindam-7 Check if the `feature` branch is still there or not. If it happens to exist, then you can delete the last commit from the `main` branch (where you've committed the changes) using: `git reset --hard HEAD~1`

> Do note that this operation is irreversible, & any changes made in the deleted commit will be **permanently** lost. So, make sure to create a backup or have a copy of the commit if you need to recover it later.
